ops/main.py

```diff
@@ -11,529 +11,18 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+"""Support legacy ops.main.main() import."""
 
-"""Main entry point to the framework.
-
-Note that this module is callable, and calls the :func:`ops.main.main` function.
-This is so that :code:`import ops` followed by :code:`ops.main(MyCharm)` works
-as expected.
-"""
-
-import logging
-import os
-import shutil
-import subprocess
-import sys
 import warnings
-from pathlib import Path
-from typing import Any, Dict, List, Optional, Tuple, Type, Union, cast
+from typing import Optional, Type
 
 import ops.charm
-import ops.framework
-import ops.model
-import ops.storage
-from ops.charm import CharmMeta
-from ops.jujucontext import _JujuContext
-from ops.log import setup_root_logging
-
-CHARM_STATE_FILE = '.unit-state.db'
-
-
-logger = logging.getLogger()
```

Comment:
For what it's worth, I'm fine with breaking people doing `from ops.main import logger`, and also things like `from ops.main import logging`.

Reply:
We could re-export symbols via new ops/main.py

Let's reach a decision on matrix.

Reply:
Yeah, I think it's reasonable to re-export symbols, but let's not get too carried away (no need to do `logger` or `os` or whatever). Just the ones people realistically might use -- though I realise that's a bit subjective.

Reply:
I've decided to re-export the set of symbols that Scenario 6 uses specifically.

Rationale: many charms use Scenario and we shouldn't potentially break a whole bunch of charms if they pin dependencies asymmetrically (compatible ops, exact scenario); and to decouple Scenario 6 update from this PR in calendar time.